[SPARK-16870][docs]Summary:add "spark.sql.broadcastTimeout" into docs/sql-programming-gu…

biglobster opened the pull request:
## What changes were proposed in this pull request?

default value for spark.sql.broadcastTimeout is 300s. and this property do not show in any docs of spark. so add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md to help people to how to fix this timeout error when it happenned
## How was this patch tested?

not need

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

…ide.md

JIRA_ID:SPARK-16870
Description:default value for spark.sql.broadcastTimeout is 300s. and this property do not show in any docs of spark. so add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md to help people to how to fix this timeout error when it happenned
Test:done

biglobster committed:
Summary:add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md
JIRA_ID:SPARK-16870
Description:default value for spark.sql.broadcastTimeout is 300s. and this property do not show in any docs of spark. so add "spark.sql.broadcastTimeout" into docs/sql-programming-guide.md to help people to how to fix this timeout error when it happenned
Test:done

srowen commented:
Many properties are not documented on purpose, because they're semi-private. What's the argument that this was intended to be public and set by callers?

biglobster commented:
I run a large number jobs with spark-sql at the same time. and meet the error that print  timeout (some job  contains the broadcast-join operator) : 
java.util.concurrent.TimeoutException: Futures timed out after [300 seconds]
        at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
        at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
        at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:107)
        at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
        at scala.concurrent.Await$.result(package.scala:107)
        at org.apache.spark.sql.execution.joins.BroadcastHashOuterJoin.doExecute(BroadcastHashOuterJoin.scala:113)
so if I reset value bigger then 300s, error do not show again

srowen commented:
Any objection to documenting it @liancheng ?

srowen commented on docs/sql-programming-guide.md:
```diff
@@ -790,6 +790,15 @@ Configuration of Parquet can be done using the `setConf` method on `SparkSession
     </p>
   </td>
 </tr>
+<tr>
```
PS this does not look like the right place for this option. This section covers Parquet. There's a later "other options" section covering broadcast join. Move it there.

biglobster replied:
done. thx :) for you suggestion

biglobster committed:
Summary:Move it to broadcast join secsion
JIRA_ID:SPARK-16870
Description:Move "spark.sql.broadcastTimeout" to broadcast join secsion
Test:done

srowen commented:
Jenkins test this please

SparkQA commented:
**[Test build #63322 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/63322/consoleFull)** for PR 14477 at commit [`52911bf`](https://github.com/apache/spark/commit/52911bf0f288270c11ba7d59ff70c459f7eabb4e).

SparkQA commented:
**[Test build #63322 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/63322/consoleFull)** for PR 14477 at commit [`52911bf`](https://github.com/apache/spark/commit/52911bf0f288270c11ba7d59ff70c459f7eabb4e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/63322/
Test PASSed.

srowen commented:
Merged to master/2.0

liancheng commented:
@srowen Sorry for the late reply, this LGTM.
